nixos/modules/services/logging/logrotate.nix

```diff
@@ -172,7 +387,16 @@ in
       serviceConfig = {
         Restart = "no";
         User = "root";
-        ExecStart = "${pkgs.logrotate}/sbin/logrotate ${configFile}";
+        ExecStart = "${pkgs.logrotate}/sbin/logrotate ${mailOption} ${cfg.configFile}";
+      };
+    };
+    systemd.services.logrotate-checkconf = {
```

Comment:
it would be better to move this to the `preStart` of the logrotate service

https://www.freedesktop.org/software/systemd/man/systemd.service.html#ExecStartPre=

> ExecStart= commands are only run after all ExecStartPre= commands that were not prefixed with a "-" exit successfully.

Reply:
I don't understand services that explicitly check their config in prestart. Start itself will do that very same check just fine, I don't see the point of moving this to preStart (well, some services could start in degraded mode with minor errors, but if so for something like logrotate I'd say we want what can be rotated rotated? That's why I didn't add a dependency on this even if this from the main service)

The reason I made this service was that I wanted a warning on configuration switch when the configuration is bad -- because logrotate.service itself was disabled in #161929 as it broke testSwitch, so we've lost this feature recently. I find it more useful to get an immediate feedback in "the following services failed to start" block of config activation than waiting 1h for the next logrotate run and notice it didn't work later (who knows when) 


If you think that's overdoing it we can just remove this extra service, though. Or maybe figure a way to re-enable the main service without making testSwitch fail...

Reply:
(Anyway, that's unrelated to the main goal of this PR, so I'd be happy to just remove it for now and discuss it elsewhere if it's debatable)